allensdk/brain_observatory/behavior/write_nwb/nwb_writer_utils.py

```diff
@@ -0,0 +1,157 @@
+import inspect
+import logging
+from typing import List
+import os
+
+from allensdk.brain_observatory.behavior.behavior_session import (
+    BehaviorSession,
+)
+from allensdk.brain_observatory.behavior.data_files.stimulus_file import (
+    BehaviorStimulusFile,
+)
+from allensdk.brain_observatory.nwb.nwb_utils import NWBWriter
+from allensdk.core.auth_config import LIMS_DB_CREDENTIAL_MAP
+from allensdk.internal.api import db_connection_creator
+from allensdk.brain_observatory.behavior.data_objects.metadata.behavior_metadata.date_of_acquisition import DateOfAcquisition  # noqa: E501
+
+
+def _compare_metadata(
+    input_id: int,
+    input_session: BehaviorSession,
+    table_metadata: dict,
+    skip_metadata: List[str],
+    skip_stim: List[str],
+):
+    """Compare data in the metadata table, loaded behavior session, and
+    the BehaviorStimulusFile to each other. Raise on conflicts.
+
+    Parameters
+    ----------
+    input_session : BehaviorSession or BehaviorOphysExperiment
+        Session object we are creating an NWB file for.
+    table_metadata : dict
+        Dictionary of data containing the metadata from the appropriate
+        metadata table.
+    skip_metadata : list of strings
+        Names of metadata keys to skip during comparison.
+    skip_stim : list of strings
+        Names of stimulus file keys to skip during comparison.
+    """
+    error_message = ""
```

Comment:
Please separate this logic into `compare_metadata_table` and `compare_stimulus_file`